src/domain/usecase/CreateSpot.ts

```diff
@@ -0,0 +1,18 @@
+import { SpotRepository } from "../repository/SpotRepository";
+import { Location } from "../model/Location";
+
+interface Dependencies {
+  spotRepository: SpotRepository;
+}
+
+export class CreateSpot {
+  private spotRepository: SpotRepository;
+
+  public constructor({ spotRepository }: Dependencies) {
+    this.spotRepository = spotRepository;
+  }
+
+  public execute(name: string, location: Location, description?: string) {
+    this.spotRepository.createSpot(name, location, description);
```

Comment:
-> and create the `CreateSpot` object here.

+ Maybe do some validation / sanitation here